### ISSUE TITLE
Consider to enable LogstashEncoder by default

### ISSUE THREAD

**Issue by skabashnyuk** [eclipse/che-lib#62]
I would like to collect opinions about changing encoder for file logs(Important note: human readable Console log will remain the same)
from 
```
        <encoder>
            <charset>utf-8</charset>
            <pattern>%-41(%date[%.15thread]) %-45([%-5level] [%.30logger{30} %L]) - %msg%n</pattern>
        </encoder>
```
to 
```
 <encoder class="net.logstash.logback.encoder.LogstashEncoder" />
```
Example not formatted 
```
{"@timestamp":"2017-11-19T20:37:00.893+00:00","@version":1,"message":"Workspace ID: workspace5pudf9crvcnrbrks","logger_name":"org.eclipse.che.
api.project.server.WorkspaceHolder","thread_name":"localhost-startStop-1","level":"INFO","level_value":20000}

```
Example formatted:
```
{
  "@timestamp": "2017-11-19T20:37:00.893+00:00",
  "@version": 1,
  "message": "Workspace ID: workspace5pudf9crvcnrbrks",
  "logger_name": "org.eclipse.che. api.project.server.WorkspaceHolder",
  "thread_name": "localhost-startStop-1",
  "level": "INFO",
  "level_value": 20000
}
```

Proc:
1. It has fixed format, will be easy to parse with different tools like ```jq```
2. ```timestamp```,```thread_name```  , ```logger_name``` are not truncated .
3. Massage takes a single line.  In some cases, if log contains multiline comment it's very hard to parse it with tools.  With json format, it won't be a problem.
4. We may consider in future using ```Logstash``` as a default log collection backend for a multi-user mode.  

Cons:
1. Larger file. I expect ~ 30%
2. Expected cons from 1) it will be a bit slower.

**Comment by sunix** [comment]
shouldn't this be an issue in http://github.com/eclipse/che ? not sure many people is following this repo

**Comment by skabashnyuk** [comment]
@riuvshin @skryzhny wdyt?

**Comment by skryzhny** [comment]
I have a doubt in old format I clearly see number of line like:
`[INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 676]`
In new proposed format I don't see it. It can be very difficult to debug without that line number.

**Comment by skabashnyuk** [comment]
@skryzhny I think it possible with 
```
<encoder class="net.logstash.logback.encoder.LogstashEncoder">
  <includeCallerData>true</includeCallerData>
</encoder>
```
see more https://github.com/logstash/logstash-logback-encoder#caller-info-fields

**Comment by jfchevrette** [comment]
+1, this would be useful to have the logs ingested and parsed by services such as logstash

**Comment by skabashnyuk** [comment]
won't fix
